### PR TITLE
Remove all non library code

### PR DESCRIPTION
**Pull request by owalk**
This pull request transitions the scope of this repository to be intended as a library game engine instead of a specific game.

The pull request does the following:
removes non-needed files from the project (to add to game/application repo)
add the tools needed for creating python wheels
adds __init__ files to all modules to allow for the inclusion of code in wheels
add setup.py for wheel creation
add build files to the gitignore

Does not:
add CI/CD tools for auto push code to public PyPi repo
Does not finish separating out lib logic from game logic

### CODE REVIEW

File: m1wengine/managers/level_manager.py [hunk 1]
```diff
@@ -81,16 +97,6 @@ def run(self) -> bool:
         if self._user_input == UserSelection.level:
             # check is menu is finished disabling
             if not self._menu.is_enabled():
```
Comment (Sean-Nishi) [inline m1wengine/managers/level_manager.py:97]: Add Todo comment about adding in method handlers to change assets like music. Similar to deleted code.
Issue here: https://github.com/owalk/Reluctant-Hero/issues/3

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"""This module is for the package setup."""
+
+from setuptools import setup, find_packages
+
+setup(
+    name="m1wengine",
+    version="1.0",
```
Comment (Sean-Nishi) [inline setup.py:7]: Prefer "1.0.0". First number is version, second number is update number, and third number is for small patches and bug fixes that don't add or change anything major

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,6 @@
 **/__pycache__/
 *.py[cod]
```
Comment (Sean-Nishi) [inline .gitignore:2]: Use ** to be consistent

**Review by Sean-Nishi — CHANGES_REQUESTED** [review]
See comments

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    description="A game engine built on top of pygame",
+    url="https://github.com/Sean-Nishi/M1Wengine",
+    packages=find_packages(),
+    install_requires=["pygame-menu==4.4.3", "pygame==2.5.2"],
```
Comment (Sean-Nishi) [inline setup.py:11]: Incorrect python version. Should be 3.11 but double check

File: m1wengine/managers/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+"""Package for m1wengine game engine class managers.
+
+These managers are used to control different facets are the game logic.
```
Comment (Sean-Nishi) [inline m1wengine/managers/__init__.py:3]: Grammar

Comment (owalk) [reply m1wengine/managers/__init__.py]: fixed

**Review by Sean-Nishi — APPROVED** [review]
Grammar and update minimum pytho

**Pull request merged**